src/quote.rs

```diff
@@ -36,13 +51,89 @@ pub struct Header {
     pub user_data: [u8; 20],
 }
 
-#[derive(Decode, Debug)]
+impl Arbitrary for Header {
```

Comment:
Code readability suffers when implementing traits manually like this, and it's error-prone. Consider using `#[derive(Arbitrary)]` instead.